File: GeneratedEntityReaderClasses/Components/TextLogComponent.cs
namespace GeneratedNoitaClasses;
public class TextLogComponent : NoitaComponentBase,  iNoitaType<TextLogComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString key { get; set; } = new();
	public NoitaString image_filename { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		key.Read(s);
		image_filename.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		key.Write(s);
		image_filename.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/TextLogComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString

from NoitaStream import NoitaStream
@dataclass
class TextLogComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    key:NoitaString = field(default_factory=NoitaString)
    image_filename:NoitaString = field(default_factory=NoitaString)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.key.Read(s)
        self.image_filename.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.key.Write(s)
        self.image_filename.Write(s)

